### PR TITLE
Migrate Intel MPI InfiniBand Test

### PR DESCRIPTION
**Pull request by kamalca**
Migrating INFINIBAND-INTEL-MPI-2VM from LISAv2.

### CODE REVIEW

File: microsoft/testsuites/hpc/infinibandsuit.py [hunk 1]
```diff
@@ -156,3 +161,91 @@ def verify_ping_pong(self, environment: Environment, log: Logger) -> None:
...
+        server_ssh.enable_public_key(client_ssh.generate_key_pairs())
+        client_ssh.enable_public_key(server_ssh.generate_key_pairs())
+        server_ssh.add_known_host(client_ip)
+        client_ssh.add_known_host(server_ip)
+
+        # Note: Using bash because script is not supported by Dash
+        # sh points to dash on Ubuntu
+        server_node.execute(
+            "bash /opt/intel/oneapi/mpi/2021.1.1/bin/mpirun "
+            f"-hosts {server_ip},{server_ip} -iface {server_dev_name} -ppn 1 -n 2 "
```
Comment (squirrelsc) [inline microsoft/testsuites/hpc/infinibandsuit.py]: Is that expected to repeat the server_ip here?

File: microsoft/testsuites/hpc/infinibandsuit.py [hunk 2]
```diff
@@ -156,3 +161,91 @@ def verify_ping_pong(self, environment: Environment, log: Logger) -> None:
...
+        run_in_parallel([server_ib.install_intel_mpi, client_ib.install_intel_mpi])
+
+        # Restart the ssh sessions for changes to /etc/security/limits.conf
+        # to take effect
+        server_node.close()
+        client_node.close()
+
+        # Get the ip adresses and device name of ib device
+        server_ib_interfaces = server_ib.get_ib_interfaces()
+        client_ib_interfaces = client_ib.get_ib_interfaces()
```
Comment (squirrelsc) [inline microsoft/testsuites/hpc/infinibandsuit.py:203]: I just noticed the return value is a tuple. How about convert it to an object with names? That's easy to use and read.

File: microsoft/testsuites/hpc/infinibandsuit.py (hunk 1 above)
Comment (kamalca) [reply microsoft/testsuites/hpc/infinibandsuit.py]: Yes, this is how it is done in LISAv2

**Comment by squirrelsc** [comment]
It looks good to me. The change of tuple to dataclass can happen before adding new test cases. That doesn't matter, but sounds better order. :)

**Review by squirrelsc — DISMISSED** [review]

**Comment by kamalca** [comment]
I reorganized the commits, but no other changes were made in the past push.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**